pkg/api/store/password/store.go

```diff
@@ -0,0 +1,276 @@
+			return fmt.Errorf("error getting secret for field %s", info.value)
+		}
+		values.PutValue(data, value, info.paths...)
+	}
+	return nil
+}
+
+func (p *PasswordStore) handlePasswordFields(fieldData []fieldInfo, data map[string]interface{}, id string) error {
+	for _, info := range fieldData {
+		var name, namespace string
+		if _, ok := data["name"]; ok {
+			name = convert.ToString(data["name"])
+		} else {
+			name = fmt.Sprintf("%s-%s", id, info.paths[len(info.paths)-1])
+		}
+		if val, ok := data["id"]; ok {
+			splitID := strings.Split(convert.ToString(val), ":")
+			if len(splitID) == 2 {
+				namespace = splitID[1]
+			} else if len(splitID) == 1 {
+				namespace = splitID[0]
+			}
+		}
+		if namespace == "" {
+			namespace = "mgmt-secrets"
+		}
+		if err := p.createOrUpdateSecrets(info.value, name, namespace); err != nil {
+			return err
+		}
+		values.PutValue(data, fmt.Sprintf("%s:%s", namespace, strings.ToLower(name)), info.paths...)
+	}
+	return nil
+}
+
+func buildFieldData(data1 map[string]interface{}, data2 map[string]interface{}, fieldData *[]fieldInfo, path []string) {
+	for key1, val1 := range data1 {
+		if val2, ok := data2[key1]; ok {
+			if convert.ToString(val1) == ".." {
+				val := convert.ToString(val2)
+				if val != "" {
+					split := strings.Split(val, ":")
+					if len(split) == 2 && (split[0] == "mgmt-secrets") {
+						// rancher referenced
+						logrus.Debugf("rancher referenced, continue %s", val)
+						continue
+					}
+				}
+				path = append(path, key1)
+				*fieldData = append(*fieldData, fieldInfo{path, val})
+			} else {
+				valArr := convert.ToMapSlice(val2)
+				if valArr == nil {
+					buildFieldData(convert.ToMapInterface(val1), convert.ToMapInterface(val2), fieldData, append(path, key1))
+				} else {
+					for _, each := range valArr {
+						buildFieldData(convert.ToMapInterface(val1), each, fieldData, append(path, key1))
+					}
+				}
+			}
+		}
+	}
+}
+
+func (p *PasswordStore) createOrUpdateSecrets(data, name, namespace string) error {
+	_, err := p.nsStore.Get(namespace, metav1.GetOptions{})
+	if err != nil && errors.IsNotFound(err) {
+		if _, err = p.nsStore.Create(&corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespace,
+			},
+		}); err != nil && !errors.IsAlreadyExists(err) {
+			return err
+		}
+	}
+	if err != nil {
+		return err
+	}
+	name = strings.ToLower(name)
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		StringData: map[string]string{name: data},
+		Type:       corev1.SecretTypeOpaque,
+	}
+	existing, err := p.secretStore.GetNamespaced(namespace, name, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		_, err = p.secretStore.Create(secret)
+		return err
+	} else if err != nil {
+		return err
+	}
+	existing.StringData = secret.StringData
+	_, err = p.secretStore.Update(existing)
+	return err
+}
+
+func (p *PasswordStore) getSecret(input []string) (string, error) {
+	returned := ""
```

Comment:
lets validate for input len here, and return empty if < 2

Reply:
didn't do it here again since is done before calling the method